enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,500 @@
+for AWS credentials should also be added to the ExternalDNS Operator. While ExternalDNS has some level of support for AWS STS, there is a
+[known caveat upstream](https://github.com/kubernetes-sigs/external-dns/issues/1944) that may affect OCP.
+
+In the event that invalid provider credentials are passed to ExternalDNS, a `ValidCredentials` status condition could be set to false on the
+ExternalDNS Operator CR. On OpenShift, this status condition could be reflected by the ExternalDNS ClusterOperator resource
+(should the ExternalDNS Operator be included in the Core OCP product).
+
+Outside of OpenShift, a user could set the `providerCredentials` field to be an object reference to a resource that holds their cloud credentials,
+such as a secret or configmap in an arbitrary namespace.
+
+By default, the operator would assume the cluster's platform is the desired DNS Provider. The operator would allow a user to specify
+an alternative External DNS Provider, such as BlueCat DNS. The ExternalDNS Operator would be able to create multiple ExternalDNS deployments
+to run in concert, one for each ExternalDNS Operator Custom Resource (similar to how the OpenShift [Ingress Operator](https://github.com/openshift/cluster-ingress-operator)
+consumes one or more Ingress Controller resources). The `domains` field would be used to filter source resources according to the requested DNS hostname.
+If an ExternalDNS Operator Custom Resource specifies multiple DNS Zones, multiple instances of ExternalDNS would be deployed within the same deployment accordingly.
+
+By default, ExternalDNS would create DNS records for all instances of the given Source resources. A union type would be defined to allow for
+source-specific parameters. In this example, ExternalDNS would create DNS Records for Route resources that match the label (read: annotation, more on this later)
+`foo: bar`, and would ignore the ExternalDNS Host Target Override Annotation (See [Implementation Details](#implementation-details) for more details about ExternalDNS
+annotations). This example would also instruct ExternalDNS to create DNS Records for CR instances of type `dnsrecords` (not a part of OCP 4.9 implementation).
+Finally, this example would instruct ExternalDNS to create DNS Records in the given Private zone for Service resource instances in namespace `foo`.
+
+ExternalDNS's `--annotation-filter` option filters source resource instances via annotations using label selector semantics, which is somewhat unexpected.
+Note that the upstream CRD source implementation actually filters based on resource labels. If upstream ExternalDNS is open to the idea of implementing
+proper label selector logic for all source types (instead of label selector based annotations), work may need to be done in the upstream ExternalDNS
+community accordingly.
+
+As a side note, ExternalDNS deployments will be configured to run ExternalDNS pods on cluster control-plane nodes.
```

Comment:
Maybe a small explanation why on the control plane nodes?

Reply:
See Dane's prior comment:
> I think it's common practice for add-on control-plane functionality to use the node roles to run on master nodes only by default.

Reply:
I'll elaborate in the EP. Thanks!